File: dp_labs_and_solutions/dp_labs_and_solutions/dp_java_labs_and_solutions/null_object/solution/Unary.java
// Unary.java

public abstract class Unary extends Operator
{
	public Unary(Expression aOperand1)
	{
		setOperand1(aOperand1);
	}

	public abstract void accept(ExpressionVisitor aExpressionVisitor);

	private void setOperand1(Expression aOperand1)
	{
		myOperand1 = aOperand1;
	}

	public Expression getOperand1()
	{
		return myOperand1;
	}

	private Expression myOperand1;
}

File: dp_labs_and_solutions/original/dp_labs_and_solutions/dp_cpp_labs_and_solutions/null_object/solution/unary.cpp
// unary.cpp

class expression;
#include "unary.h"

unary::unary(expression * a_operand1)
{
	set_operand1(a_operand1);
}

void unary::set_operand1(expression * a_operand1)
{
	my_operand1 = a_operand1;
}

expression * unary::get_operand1() const
{
	return my_operand1;
}
